crates/syntax/src/algo.rs

```diff
@@ -289,11 +289,19 @@ fn _replace_children(
     with_children(parent, new_children)
 }
 
+#[derive(Debug, PartialEq, Eq, Hash)]
+enum InsertPos {
+    FirstChildOf(SyntaxNode),
+    Before(SyntaxElement),
+    After(SyntaxElement),
```

Comment:
For public insert position, we need to suport both before and after for nicer API. But I wonder if this internal type can get by only with `After`?

Reply:
If we can figure out whether something is the first child of a node or not we could I imagine. But I don't know how I could possibly check that.

Reply:
`it.prev_sibling().is_none()`

Reply:
Okay I managed to remove it